artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/Upgrade.java

```diff
@@ -0,0 +1,318 @@
+                     throw new RuntimeException("Script artemis was manually updated. Can't proceed with upgrade");
+                  }
+
+                  nextLine = iterator.next().trim();
+                  if (!nextLine.startsWith("fi")) {
+                     throw new RuntimeException("Script artemis was manually updated. Can't proceed with upgrade");
+                  }
+                  return null;
+               }
+
+               case "if [ -z \"$LOG_MANAGER\" ] ; then":
+                  return null;
+               case "# this is the one found when the server was created": {
+                  // looking for the following snippet:
+                  // https://github.com/apache/activemq-artemis/blob/fb1b362b473cad51ae5d05a897be02b1fa8461d4/artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/bin/artemis#L103-L109
+                  String nextLine = iterator.next().trim();
+                  if (!nextLine.startsWith("LOG_MANAGER=")) {
+                     throw new RuntimeException("Script artemis was manually updated. Can't proceed with upgrade");
+                  }
+                  nextLine = iterator.next().trim();
+                  if (!nextLine.startsWith("fi")) {
+                     throw new RuntimeException("Script artemis was manually updated. Can't proceed with upgrade");
+                  }
+                  return null;
+               }
+
+               case "# Set Defaults Properties":
+               case "ARTEMIS_LOGGING_CONF=\"$ARTEMIS_INSTANCE_ETC_URI/logging.properties\"":
+               case "ARTEMIS_LOG_MANAGER=org.jboss.logmanager.LogManager":
+               case "# finding the Log Manager":
+               case "LOG_MANAGER=`ls $ARTEMIS_HOME/lib/jboss-logmanager*jar 2>/dev/null`":
+               case "WILDFLY_COMMON=`ls $ARTEMIS_HOME/lib/wildfly-common*jar 2>/dev/null`":
+                  return null;
+            }
+            return input;
+         });
+      }
+   }
+
+   private void upgradeArtemisCmd(File bkpFolder, File bin, File etc) throws Exception {
+      File artemis = new File(bin, "artemis.cmd");
+      if (artemis.exists()) {
+         File artemisCopy = new File(bkpFolder, "artemis.cmd");
+         Files.copy(artemis.toPath(), artemisCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+         System.out.println("Converting artemis script. Original file backed up at " + bkpFolder.getAbsolutePath());
+
+         upgrade(artemisCopy, artemis, (input, iterator, out) -> {
+            switch (input.trim()) {
+               case "rem \"Set Defaults.\"":
+               case "set ARTEMIS_LOGGING_CONF=%ARTEMIS_INSTANCE_ETC_URI%/logging.properties":
+               case "set ARTEMIS_LOG_MANAGER=org.jboss.logmanager.LogManager":
+               case "set JVM_ARGS=%JVM_ARGS% -Djava.util.logging.manager=%ARTEMIS_LOG_MANAGER%":
+               case " set JVM_ARGS=%JVM_ARGS% -Dlogging.configuration=%ARTEMIS_LOGGING_CONF%":
```

Comment:
Space at the beginning of this line, but the input is trimmed, won't ever match

Reply:
thanks!